From 3af3dfb4820755168ad55fa9e494883c7170a7bc Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Sat, 28 Apr 2018 23:40:41 +0100
Subject: [PATCH 11/12] DcTracker: Change access modifier of isNvSubscription
 to protected

 * This fixes the fatal exception that makes telephony crash on CDMA networks:

     E AndroidRuntime: java.lang.IllegalAccessError: Method 'boolean com.android.internal.telephony.dataconnection.DcTracker.isNvSubscription()' is inaccessible to class 'com.qualcomm.qti.internal.telephony.dataconnection.QtiDcTracker' (declaration of 'com.qualcomm.qti.internal.telephony.dataconnection.QtiDcTracker' appears in /system/framework/qti-telephony-common.jar)

REGRESSION-1912

Change-Id: Ice02949e4114b969ad639ffd12b61579c265f40c
---
 .../android/internal/telephony/dataconnection/DcTracker.java    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index f0a8fb6ad..3107cda2c 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -1216,7 +1216,7 @@ public class DcTracker extends Handler {
         setupDataOnConnectableApns(Phone.REASON_DATA_ATTACHED);
     }
 
-    private boolean isNvSubscription() {
+    protected boolean isNvSubscription() {
         int cdmaSubscriptionSource = CdmaSubscriptionSourceManager.getDefault(mPhone.getContext());
         return cdmaSubscriptionSource == CdmaSubscriptionSourceManager.SUBSCRIPTION_FROM_NV;
     }
-- 
2.17.1

